src/Servers/HttpSys/src/RequestProcessing/Request.cs

```diff
@@ -66,19 +67,26 @@ internal Request(RequestContext requestContext, NativeRequestContext nativeReque
                 PathBase = string.Empty;
                 Path = string.Empty;
             }
-            // These paths are both unescaped already.
-            else if (originalPath.Length == prefix.Path.Length - 1)
+            else if (prefix == null)
             {
-                // They matched exactly except for the trailing slash.
-                PathBase = originalPath;
-                Path = string.Empty;
+                // Url registrations might be out of sync between instances sharing a queue.
+                PathBase = string.Empty;
+                Path = originalPath;
             }
-            else
+            // Make sure it was really a match. The queue configuration may be out of sync with other instances.
+            else if (prefix.PathWithoutTrailingSlash.Length > 0
```

Comment:
Nevermind, this logic is wrong. We can only rely on UrlContext if this instance created the queue. Otherwise we don't know what those indexes were set to. We'll have to iterate through the collection and look for the longest match manually.